server/cluster_worker.go

```diff
@@ -294,68 +40,44 @@ func (c *raftCluster) handleAddPeerReq(region *metapb.Region) (*metapb.Peer, err
 	// check this.
 	// 2, We can check the store statistics and find a low load store.
 	// 3, more algorithms...
-	var matchStore *metapb.Store
+L:
 	for _, store := range mu.stores {
-		storeID := store.GetId()
-
-		existStore := false
+		// we can't add peer in the same store.
 		for _, peer := range region.Peers {
-			if peer.GetStoreId() == storeID {
-				// we can't add peer in the same store.
-				existStore = true
-				break
+			if peer.GetStoreId() == store.GetId() {
+				continue L
 			}
 		}
-
-		if existStore {
-			continue
-		}
-
-		matchStore = &store
-
-		break
-	}
-
-	if matchStore == nil {
-		return nil, errors.Errorf("find no store to add peer for region %v", region)
-	}
-
-	peer := &metapb.Peer{
-		StoreId: proto.Uint64(matchStore.GetId()),
-		Id:      proto.Uint64(peerID),
+		return &metapb.Peer{
+			Id:      proto.Uint64(peerID),
+			StoreId: proto.Uint64(store.GetId()),
+		}, nil
 	}
-
-	return peer, nil
+	return nil, errors.Errorf("find no store to add peer for region %v", region)
 }
 
 // If leader is nil, we will return an error, or else we can remove none leader peer.
-func (c *raftCluster) handleRemovePeerReq(region *metapb.Region, leader *metapb.Peer) (*metapb.Peer, error) {
+func (c *raftCluster) handleRemovePeerReq(region *metapb.Region, leaderID uint64) (*metapb.Peer, error) {
 	if len(region.Peers) <= 1 {
 		return nil, errors.Errorf("can not remove peer for region %v", region)
 	}
-	if leader == nil {
-		return nil, errors.Errorf("invalid leader for region %v", region)
-	}
-
 	for _, peer := range region.Peers {
-		if peer.GetId() != leader.GetId() {
+		if peer.GetId() != leaderID {
 			return peer, nil
 		}
 	}
-
 	// Maybe we can't enter here.
 	return nil, errors.Errorf("find no proper peer to remove for region %v", region)
 }
 
-func (c *raftCluster) HandleAskChangePeer(request *pdpb.AskChangePeerRequest) error {
+func (c *raftCluster) handleChangePeerReq(region *metapb.Region, leaderID uint64) (*pdpb.ChangePeer, error) {
 	clusterMeta, err := c.GetConfig()
 	if err != nil {
-		return errors.Trace(err)
+		return nil, errors.Trace(err)
 	}
 
 	var (
 		maxPeerNumber = int(clusterMeta.GetMaxPeerNumber())
```

Comment:
s/maxPeerNumber/maxPeerCount/g
we should change all peer number to peer count except kvproto now.